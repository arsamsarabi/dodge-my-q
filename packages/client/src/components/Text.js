/*
  eslint
    "react/destructuring-assignment": "off",
    "no-nested-ternary": "off"
*/
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

const H1 = styled.h1`
  color: ${props => props.color};
  font-family: 'Press Start 2P', Arial, Helvetica, sans-serif;
  font-size: 2em;
  line-height: 1.5;
  font-weight: ${props => props.weight};
  margin: 0;
  transition: all 300ms ease;
  ${props => props.underline && css`
    border-bottom: 1px 
      ${props.underlineStyle === 'dashed' ? 'dashed'
    : props.underlineStyle === 'dotted' ? 'dotted'
      : 'solid'}
      ${props.color || props.theme.text};
  `}
`

H1.propTypes = {
  color: PropTypes.string,
  weight: PropTypes.number,
  underline: PropTypes.bool,
  underlineStyle: PropTypes.oneOf(['solid', 'dashed', 'dotted']),
}

H1.defaultProps = {
  color: '#333',
  weight: 400,
  underline: false,
  underlineStyle: 'solid',
}

const H2 = H1.withComponent('h2').extend`
  font-size: 1.5em;
`
const H3 = H1.withComponent('h3').extend`
  font-size: 1.3em;
`
const H4 = H3.withComponent('h4').extend`
  font-size: 1em;
`
const H5 = H3.withComponent('h5').extend`
  font-size: 0.8em;
`
const H6 = H3.withComponent('h6').extend`
  font-size: 0.7em;
`
const Text = H1.withComponent('p').extend`
  font-family: 'Roboto', Arial, Helvetica, sans-serif;
  font-size: 1em;
`

export default Text
export { H1, H2, H3, H4, H5, H6, Text }
