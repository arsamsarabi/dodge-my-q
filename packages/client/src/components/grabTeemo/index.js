/*
  eslint
    "filenames/match-exported": "off",
    "import/extensions": "off"
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import GrabImage from 'resources/images/grab.png';

const Wrapper = styled.div`
  position: fixed;
  bottom: 0;
  left: 0;
  width: 200px;

`;

const GrabTeemo = props => (
  <Wrapper>
    <img src={GrabImage} alt="Blitzcrank Grabbing Teemo xd"/>
  </Wrapper>
);

GrabTeemo.propTypes = {

};

export default GrabTeemo;
