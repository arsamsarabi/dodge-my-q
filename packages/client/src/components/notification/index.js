import React from 'react'
import Reflux from 'reflux'
import Snackbar from '@material-ui/core/Snackbar'
import IconButton from '@material-ui/core/IconButton'
import CloseIcon from '@material-ui/icons/Close'
import LeagueStore from 'stores/league/store'
import LeagueActions from 'stores/league/actions'


export default class Notifications extends Reflux.Component {
  constructor(props) {
    super(props)
    this.state = {}
    this.store = LeagueStore
    this.storeKeys = [
      'notificationOpen',
      'notificationMessage',
    ]
  }
  render() {
    const {
      notificationOpen,
      notificationMessage,
    } = this.state
    return <Snackbar
      anchorOrigin={{ vertical: 'top', horizontal: 'left' }}
      open={notificationOpen}
      ContentProps={{
        'aria-describedby': 'message-id',
      }}
      autoHideDuration={6000}
      message={<span id="message-id">{notificationMessage}</span>}
      action={[
        <IconButton
          key="close"
          aria-label="Close"
          color="inherit"
          onClick={() => LeagueActions.dismissNotification()}
        >
          <CloseIcon />
        </IconButton>,
      ]}
    />
  }
}