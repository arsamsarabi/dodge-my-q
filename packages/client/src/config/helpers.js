/*
  eslint
    "camelcase": "off",
    "complexity": "off",
*/

import {
  bronze_i,
  bronze_ii,
  bronze_iii,
  bronze_iv,
  bronze_v,
  silver_i,
  silver_ii,
  silver_iii,
  silver_iv,
  silver_v,
  gold_i,
  gold_ii,
  gold_iii,
  gold_iv,
  gold_v,
  platinum_i,
  platinum_ii,
  platinum_iii,
  platinum_iv,
  platinum_v,
  diamond_i,
  diamond_ii,
  diamond_iii,
  diamond_iv,
  diamond_v,
  provisional,
  bronze,
  silver,
  gold,
  platinum,
  diamond,
  master,
  challenger,
} from './leagueImages'

const helpers = {}

const urlFriendly = username => username.toLowerCase().replace(/\s/g, '')
helpers.urlFriendly = urlFriendly

const getLeagueImage = (tier, rank) => {
  if (tier === 'CHALLENGER') {
    return challenger
  } else if (tier === 'MASTER') {
    return master
  } else if (tier === 'DIAMOND') {
    switch (rank) {
    case 'I':
      return diamond_i
    case 'II':
      return diamond_ii
    case 'III':
      return diamond_iii
    case 'IV':
      return diamond_iv
    case 'V':
      return diamond_v
    default:
      return diamond
    }
  } else if (tier === 'PLATINUM') {
    switch (rank) {
    case 'I':
      return platinum_i
    case 'II':
      return platinum_ii
    case 'III':
      return platinum_iii
    case 'IV':
      return platinum_iv
    case 'V':
      return platinum_v
    default:
      return platinum
    }
  } else if (tier === 'GOLD') {
    switch (rank) {
    case 'I':
      return gold_i
    case 'II':
      return gold_ii
    case 'III':
      return gold_iii
    case 'IV':
      return gold_iv
    case 'V':
      return gold_v
    default:
      return gold
    }
  } else if (tier === 'SILVER') {
    switch (rank) {
    case 'I':
      return silver_i
    case 'II':
      return silver_ii
    case 'III':
      return silver_iii
    case 'IV':
      return silver_iv
    case 'V':
      return silver_v
    default:
      return silver
    }
  } else if (tier === 'BRONZE') {
    switch (rank) {
    case 'I':
      return bronze_i
    case 'II':
      return bronze_ii
    case 'III':
      return bronze_iii
    case 'IV':
      return bronze_iv
    case 'V':
      return bronze_v
    default:
      return bronze
    }
  }
  return provisional
}
helpers.getLeagueImage = getLeagueImage

export default helpers
export {
  urlFriendly,
  getLeagueImage,
}
