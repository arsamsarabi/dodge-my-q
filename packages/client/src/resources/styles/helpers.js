const helpers = {};

const pageContainer = () => `
  min-height: calc(100vh - 50px);
  height: calc(100vh - 50px);
  width: 100vw;
  max-width: 1024px;
  margin: 0 auto;
  overflow-y: auto;
  padding: 4px 24px;
`

helpers.pageContainer = pageContainer

export default helpers
export {
  pageContainer,
}

