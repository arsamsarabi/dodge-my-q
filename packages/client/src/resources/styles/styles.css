@import url('https://fonts.googleapis.com/css?family=Press+Start+2P|Roboto');

* {
	margin: 0;
	padding: 0;
	box-sizing: border-box;
}

html, body {
	width: 100vw;
	min-height: 100vh;
}

body {
  font-family: 'Roboto', Arial;
	font-size: 18px;
	overflow: hidden;
}

img {
	max-width: 100%;
	height: auto;
}

/* Scrollbar Chrome only */
::-webkit-scrollbar {
	width: 10px;
}
::-webkit-scrollbar-track {
  background: #b2bec3;
  transition: 0.2s all linear ;
}
::-webkit-scrollbar-track:hover {
	background: #F2F1EF;
  transition: 0.2s all linear ;
}
::-webkit-scrollbar-thumb {
	background: #0967b3;
  transition: 0.2s all linear ;
  border-radius: 5px;
}
::-webkit-scrollbar-thumb:hover {
	background: #636e72;
  transition: 0.2s all linear ;
}