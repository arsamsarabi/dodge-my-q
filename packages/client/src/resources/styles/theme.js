import colors from './colors'

const theme = {
  background: colors.white,
  darkBackground: colors.black,
  text: colors.black,
  lightText: colors.white,
  primary: colors.purple,
  secondary: colors.secondary,
  accent: colors.blue,
  success: colors.success,
  warning: colors.warning,
  danger: colors.danger,
  darkGrey: colors.darkGrey,
  grey: colors.grey,
  lightGrey: colors.lightGrey,
  blue:colors.primary,
  red: colors.danger,
  white: colors.white,
}

export { theme }
export default theme
