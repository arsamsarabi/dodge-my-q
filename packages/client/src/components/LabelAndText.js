import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Text } from './Text';

const Wrapper = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  text-align: left;
  /* width: 100%; */
  & > p:first-of-type {
    flex: 1;
  }
`;

const LabelAndText = ({ label, text }) => (
  <Wrapper>
    <Text weight={600}>
      {label}
    </Text>
    <Text>
      {text}
    </Text>
  </Wrapper>
);

LabelAndText.propTypes = {
  label: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
};

export default LabelAndText;
