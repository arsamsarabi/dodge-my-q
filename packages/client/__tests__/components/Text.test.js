import 'jest-styled-components'
import React from 'react'
import { H1, H2, H3, H4, H5, H6, Text } from '../../src/components/Text'
import { renderWithTheme } from '../helpers'
import { theme } from '../../src/resources/styles/theme'

describe('Heading element', () => {
  it('Should render a Text element', () => {
    const component = renderWithTheme(<Text>
      Text Paragraph
    </Text>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render a Text element with theme', () => {
    const component = renderWithTheme(<Text underline theme={theme}>
      Themed Text Paragraph
    </Text>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render a Text element without theme', () => {
    const component = renderWithTheme(<Text underline theme={{}}>
      Not themed Text Paragraph
    </Text>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render a H1 element', () => {
    const component = renderWithTheme(<H1>
      H1 Heading
    </H1>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render a H2 element', () => {
    const component = renderWithTheme(<H2>
      H2 Heading
    </H2>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render a H3 element', () => {
    const component = renderWithTheme(<H3>
      H3 Heading
    </H3>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render a H4 element', () => {
    const component = renderWithTheme(<H4>
      H4 Heading
    </H4>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render a H5 element', () => {
    const component = renderWithTheme(<H5>
      H5 Heading
    </H5>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render a H6 element', () => {
    const component = renderWithTheme(<H6>
      H6 Heading
    </H6>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render an underlined heading element', () => {
    const component = renderWithTheme(<H1 underline>
      Underlined H1 Heading
    </H1>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render an underlined dotted heading element', () => {
    const component = renderWithTheme(<H1 underline underlineStyle="dotted">
      Underlined Dotted H1 Heading
    </H1>).toJSON()
    expect(component).toMatchSnapshot()
  })
  it('Should render an underlined dashed heading element', () => {
    const component = renderWithTheme(<H1 underline underlineStyle="dashed">
      Underlined Dashed H1 Heading
    </H1>).toJSON()
    expect(component).toMatchSnapshot()
  })
})
