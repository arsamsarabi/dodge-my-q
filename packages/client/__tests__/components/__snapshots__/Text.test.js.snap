// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading element Should render a H1 element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
}

<h1
  className="c0"
  color="#333"
>
  H1 Heading
</h1>
`;

exports[`Heading element Should render a H2 element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  font-size: 1.5em;
}

<h2
  className="c0"
  color="#333"
>
  H2 Heading
</h2>
`;

exports[`Heading element Should render a H3 element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  font-size: 1.3em;
}

<h3
  className="c0"
  color="#333"
>
  H3 Heading
</h3>
`;

exports[`Heading element Should render a H4 element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  font-size: 1.3em;
  font-size: 1em;
}

<h4
  className="c0"
  color="#333"
>
  H4 Heading
</h4>
`;

exports[`Heading element Should render a H5 element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  font-size: 1.3em;
  font-size: 0.8em;
}

<h5
  className="c0"
  color="#333"
>
  H5 Heading
</h5>
`;

exports[`Heading element Should render a H6 element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  font-size: 1.3em;
  font-size: 0.7em;
}

<h6
  className="c0"
  color="#333"
>
  H6 Heading
</h6>
`;

exports[`Heading element Should render a Text element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  font-family: 'Roboto',Arial,Helvetica,sans-serif;
  font-size: 1em;
}

<p
  className="c0"
  color="#333"
>
  Text Paragraph
</p>
`;

exports[`Heading element Should render a Text element with theme 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  border-bottom: 1px  solid #333;
  font-family: 'Roboto',Arial,Helvetica,sans-serif;
  font-size: 1em;
}

<p
  className="c0"
  color="#333"
>
  Themed Text Paragraph
</p>
`;

exports[`Heading element Should render a Text element without theme 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  border-bottom: 1px  solid #333;
  font-family: 'Roboto',Arial,Helvetica,sans-serif;
  font-size: 1em;
}

<p
  className="c0"
  color="#333"
>
  Not themed Text Paragraph
</p>
`;

exports[`Heading element Should render an underlined dashed heading element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  border-bottom: 1px  dashed #333;
}

<h1
  className="c0"
  color="#333"
>
  Underlined Dashed H1 Heading
</h1>
`;

exports[`Heading element Should render an underlined dotted heading element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  border-bottom: 1px  dotted #333;
}

<h1
  className="c0"
  color="#333"
>
  Underlined Dotted H1 Heading
</h1>
`;

exports[`Heading element Should render an underlined heading element 1`] = `
.c0 {
  color: #333;
  font-size: 2em;
  line-height: 1.5;
  font-weight: 400;
  margin: 0;
  -webkit-transition: all 300ms ease;
  transition: all 300ms ease;
  border-bottom: 1px  solid #333;
}

<h1
  className="c0"
  color="#333"
>
  Underlined H1 Heading
</h1>
`;
